### PR TITLE
Restrict mac codesign and device_doctor builds to device:none hosts

### PR DESCRIPTION
**Pull request by yusuf-goog**
Bug:https://github.com/flutter/flutter/issues/119366

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by nehalvpatel — APPROVED** [review]
LGTM!

File: .ci.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ targets:
       add_recipes_cq: "true"
       script: device_doctor/tool/build.sh
       cipd_name: flutter/device_doctor/mac-amd64
+      device_type: none
```
Comment (ricardoamador) [inline .ci.yaml:45]: Don't you also have to change the platform for this to take effect? I have had this issue with emulators where it needs to run on a linux platform and not linux_android platform.

Comment (yusuf-goog) [reply .ci.yaml:45]: Mac doesn't specify the device type here? so it should be ok?

Comment (ricardoamador) [reply .ci.yaml:45]: Ah yeah I see that. You are good.

**Review by ricardoamador — APPROVED** [review]
LGTM

**Pull request merged**